### DISCUSSION TITLE
How to prevent Avalonia from scanning for AXAML files in hidden folders

### DISCUSSION THREAD

**Discussion by yaroslav-khomyak** [AvaloniaUI/Avalonia#13723] [Q&A] [answered]
I want to use VSHistory plugin for local file history, as an additional security layer over Git.

VSHistory creates hidden .vshistory folder, which contains datastamped copys of source files.
It works fine with .cs files, and with Git.

But, AXAML compiler is seeing those hidden timestamped files, and tries to compile them. With throwing bunch of 'duplicated thing' errors.

Please help me with workaround.

**Comment by timunie** [comment]
need to exclude them from the project on *.csproject-file

**Reply by yaroslav-khomyak** [reply]
I've tried `<Compile Remove="**\.vshistory\**\*.axaml" />` with no luck.
Could you point me to the right direction?

**Reply by jp2masa** [reply, marked as answer]
It's `AvaloniaXaml`:

https://github.com/AvaloniaUI/Avalonia/blob/59d44e97bfab65227b0df9c231b2bef03443dca9/packages/Avalonia/AvaloniaBuildTasks.props#L8

**Reply by jp2masa** [reply]
Maybe Avalonia should add `Exclude="$(DefaultItemExcludes);$(DefaultExcludesInProjectFolder)"` like .NET SDK:

https://github.com/dotnet/sdk/blob/e80b07ab62f322e8adef5d32bae2026a2a2bd813/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.DefaultItems.props#L30

**Reply by yaroslav-khomyak** [reply]
`<AvaloniaXaml Remove="**\.vshistory\**" />` worked.  Thank you.

I guess it should be raised as an issue, because compiling files from hidden folder started with a dot is not a good idea anyway.

**Reply by timunie** [reply]
There may be ppl wanting this. In Windows, a . does not mean hidden...

**Reply by jp2masa** [reply]
I don't know how vshistory works, but I would guess it marks the folder as hidden on Windows. Anyway, I think Avalonia should do the same as the .NET SDK for default includes, as pointed [above](https://github.com/AvaloniaUI/Avalonia/discussions/13723#discussioncomment-7663375). I can create a PR for that if you agree.

**Reply by yaroslav-khomyak** [reply]
Vs history creates .vshistory folders (marked as hidden) in each folder with source code. 
.NET compiler ignores those folders by default. The issue is with .axaml files only.

**Reply by timunie** [reply]
I have no strong feelings here. But I guess it may count as a breaking change. So we may need to postpone this to the 12.o time line. Opening a feature request to track it in that case may be needed.
